src/navigation/reroute.js

```diff
@@ -275,7 +281,6 @@ export function reroute(pendingPromises = [], eventArguments) {
         originalEvent: eventArguments?.[0],
         oldUrl,
         newUrl,
-        navigationIsCanceled,
```

Comment:
Removing this property is a breaking change. I'd rather avoid a breaking change for this feature, if possible, but am not sure how to do so with async cancellation since the events fire synchronously before the async cancellation promises have resolved. Ideas?

Reply:
Yeah, I don't have a good answer for this one. If I'm understanding the current code correctly, the flag will be set only on the new cancellation event, right? If we are able to tag the cancellation navigation, then I think we could add `navigationIsCanceled: false` to the extraProperties when we create the event.